pkg/build/api/v1beta3/types.go

```diff
@@ -85,6 +85,9 @@ type BuildStatus struct {
 
 	// Config is an ObjectReference to the BuildConfig this Build is based on.
 	Config *kapi.ObjectReference `json:"config,omitempty"`
+
+	// Revision is the revision that was built with this build
+	Revision *SourceRevision `json:"revision,omitempty" description:"source revision built with this build"`
 }
```

Comment:
actually, I'm confused now... it looks like there are two places in the spec to set a commit:
```
spec.revision.git.commit...
spec.source.git.ref (could point to a branch, tag, or specific commit)
```
what happens today if those disagree?